test/e2e/network/service.go

```diff
@@ -3119,7 +3119,6 @@ var _ = common.SIGDescribe("Services", func() {
 
 			execHostnameTest(*pausePod0, nodePortAddress0, webserverPod0.Name)
 			execHostnameTest(*pausePod1, nodePortAddress0, webserverPod0.Name)
-			execHostnameTest(*pausePod1, nodePortAddress1, webserverPod0.Name)
```

Comment:
we need to update the comments above to explain what is going on, and also mention that this case is not tested with references to the discussions

Reply:
ok, so this assumes that  a pod serving in in nodeA should be reachable from a pod in nodeB that connectes to a nodePort in the same node .... 😄  a bit of a stretch I'm fine with this change, seems an artifact of kube-proxy implementation and at minimum debatable from a more fine interpretation

Reply:
I need to do some more updates to https://github.com/kubernetes-sigs/kpng/blob/master/doc/service-proxy.md. It mentions the LB IP case but doesn't say anything about NodePorts+eTP:Local.

Anyway, I agree that this is testing an artifact of kube-proxy's implementation that I don't think we actually agreed was supposed to be the behavior for _all_ service proxy implementations. (Which means maybe we should make nftables _not_ do it that way?)

Note that we already have [a unit test for pod-to-eTP:Local-NodePort in iptables](https://github.com/kubernetes/kubernetes/blob/v1.29.0/pkg/proxy/iptables/proxier_test.go#L5582-L5596) that asserts the same behavior as seen here, so removing this test will not leave us in danger of regressing that.

Reply:
Updated the comments, let me know what you think.